pkg/registry/core/service/strategy.go

```diff
@@ -120,7 +120,11 @@ func (svcStrategy) AllowUnconditionalUpdate() bool {
 //	    newSvc.Spec.MyFeature = nil
 //	}
 func dropServiceDisabledFields(newSvc *api.Service, oldSvc *api.Service) {
-
+	// Drop condition for TrafficDistribution field.
+	isTrafficDistributionInUse := (oldSvc != nil && oldSvc.Spec.TrafficDistribution != nil && *oldSvc.Spec.TrafficDistribution != "")
```

Comment:
nil and "" are not equivalent

Reply:
Good catch, since this is a pointer, we just need to allow nil as the empty value and reject "".

Reply:
Nice thanks! Changed.

(Now I'm wondering why was a trying to treat them the same, when the entire point with pointers was to distinguish nil and empty)